apps/server/src/modules/authentication/controllers/api-test/logout.api.spec.ts

```diff
@@ -69,4 +85,347 @@ describe('Logout Controller (api)', () => {
 			});
 		});
 	});
+
+	describe('externalSystemLogout', () => {
+		let currentUser: ICurrentUser;
+
+		const setupTestWithMocks = async (mockHttpPostSuccess = true) => {
+			const postMockReturnValue = mockHttpPostSuccess
+				? of(axiosResponseFactory.build())
+				: throwError(() => axiosErrorFactory.build());
+
+			const moduleFixture: TestingModule = await Test.createTestingModule({
+				imports: [ServerTestModule],
+			})
+				.overrideGuard(JwtAuthGuard)
+				.useValue({
+					canActivate(context: ExecutionContext) {
+						const req: Request = context.switchToHttp().getRequest();
+						req.user = currentUser;
+						return true;
+					},
+				})
+				.overrideProvider(HttpService)
```

Comment:
There is an AxiosMock library in the sever that you can use to mock outgoing requests